cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -95,6 +99,29 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 			NCVersion: int(nc.Version),
 		}
 	}
+
+	// Add IPs from CIDR block to the secondary IPConfigs
+	if nc.Type == v1alpha.VNETBlock {
+		// Delete primary IP reserved for Primary IP for NC
+		delete(secondaryIPConfigs, primaryPrefix.Addr().String())
+
+		for _, ipAssignment := range nc.IPAssignments {
+			cidrPrefix, err := netip.ParsePrefix(ipAssignment.IP)
+			if err != nil {
+				return nil, errors.Wrapf(ErrInvalidCIDRBlock, "CIDR Block: %s", ipAssignment.IP)
+			}
```

Comment:
it looks like this `err` is not included in the return, so we are losing the cause of the failure

Reply:
updated